### DISCUSSION TITLE
[Feature] Search by filename

### DISCUSSION THREAD

**Discussion by johanjongsma** [immich-app/immich#7937] [Feature Request]
### The feature

I would like to be able to search for a [part of] the original filename, preferably not case-sensitive, as extensions can be finicky, and with support for 'special characters' like '-1' or '(1)'. If we would go for regex, simple searches could easily become very complicated.

What do you think?

Johan

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by alextran1502** [comment]
You can already search for file name on the web, we are bringing it to the mobile app soon.

![image](https://github.com/immich-app/immich/assets/27055614/1037ea01-d665-4419-a8b2-2f43db829169)

**Reply by johanjongsma** [reply]
I understand, but why not here? It feels more intuitive
![image](https://github.com/immich-app/immich/assets/44827424/84e40ee8-e3d2-4b8d-9659-009477478912)

**Reply by bo0tzz** [reply]
Because supporting arbitrary search queries in a free-text field is really complicated ;) For example, how would you indicate whether you want a context search or a filename search?

**Reply by johanjongsma** [reply]
Why not both? Just like Google Photos? While we are at it: I would like to be able to search for 'Johan beach', which would show photos of me on the beach. Or 'johan alex', showing photos on which we are both.

**Reply by bo0tzz** [reply]
> Why not both? 

Because like I mentioned that's really complicated

**Reply by johanjongsma** [reply]
That's not a reason. This project is awesome as it is; not because they went the 'easy' way, but because they saw it was very difficult and did it anyway. How can I contribute? I'm no expert in node.js at all, but am very experienced in querying.

**Reply by alextran1502** [reply]
Probably an enhancement to the feature. As we want to push out usable features to serve the users first, iteratively improving each feature is a way to go. We talked about building our own query language, but it is a nice to have addition, so we will look it into it in the future.
